ci/kokoro/docker/e2e/Dockerfile

```diff
@@ -0,0 +1,152 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM ubuntu:18.04
+
+# Dependencies
+RUN \
+  apt-get update && \
+  apt-get -y install \
+          autoconf \
+          build-essential \
+          clang-tidy \
+          cmake \
+          curl \
+          gcc \
+          git-core \
+          golang \
+          g++ \
+          libtool \
+          libssl-dev \
+          make \
+          pkg-config \
+          wget \
+          vim \
+          zip \
+          zlib1g-dev
+
+# Build directory
+RUN mkdir -p /build && \
+    chmod 0755 /build
+
+WORKDIR /build
+
+# Install c-ares
+RUN \
+  wget -q https://github.com/c-ares/c-ares/archive/cares-1_15_0.tar.gz && \
+  tar -xf cares-1_15_0.tar.gz && \
+  cd c-ares-cares-1_15_0 && \
+  mkdir build && cd build && \
+  cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr/local .. && \
+  make install && \
+  ldconfig
+
+# Install protobuf
+RUN \
+  wget -q https://github.com/google/protobuf/archive/v3.11.2.tar.gz && \
+  tar -xf v3.11.2.tar.gz && \
+  cd protobuf-3.11.2/cmake && \
+  cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DBUILD_SHARED_LIBS=yes \
+        -Dprotobuf_BUILD_TESTS=OFF \
+        -H. -Bcmake-out && \
+    cmake --build cmake-out -- -j ${NCPU:-4} && \
+    cmake --build cmake-out --target install -- -j ${NCPU:-4} && \
+    ldconfig
+
+# Install gRPC and abseil
+RUN \
+  git clone https://github.com/grpc/grpc.git && \
```

Comment:
Why are we using `git`?  Downloading the tarball is usually faster.

Reply:
Done